openlogin/authorized-apps/index.test.ts

```diff
@@ -0,0 +1,157 @@
+    const page2 = await context2.newPage();
+    await page2.goto(href);
+    await page2.waitForSelector(
+      "text=Close this and return to your previous window",
+      {
+        timeout: 10000,
+      }
+    );
+    await page2.close();
+    await mailosaur.messages.deleteAll(process.env.MAILOSAUR_SERVER_ID || ""); // Deleting emails in email server.
+
+    await useAutoCancelShareTransfer(page);
+    await useAutoCancel2FASetup(page);
+    await page.waitForURL(`${openloginURL}/wallet/home`, {
+      waitUntil: "load",
+    });
+  });
+  test.afterAll(async ({ browser }) => {
+    browser.close;
+  });
+  test(`should display "You are not connected to any applications yet." when no apps are connected.`, async ({
+    openloginURL,
+  }) => {
+    await page.goto(`${openloginURL}/wallet/apps`);
+    await page.waitForURL(`${openloginURL}/wallet/apps`, {
+      waitUntil: "load",
+    });
+    expect(page.url()).toBe(`${openloginURL}/wallet/apps`);
+    expect(
+      await page.isVisible(
+        "text=You are not connected to any applications yet."
+      )
+    ).toBeTruthy();
+  });
+
+  //below test use testEmail to login to solana wallet prod and there by generates an app share
+  // which gets listed in list of apps
+  test(`should login to solana wallet with passwordless login and list app`, async ({
+    openloginURL,
+    browser,
+  }) => {
+    const context2 = await browser.newContext();
+    const context3 = await browser.newContext();
+    const page2 = await context2.newPage();
+    await page2.goto("https://solana.tor.us/login");
+    await page2.fill('[placeholder="Enter your email"]', testEmail);
+    await page2.click('button:has-text("Continue with Email")');
+    await page.waitForTimeout(4000);
+
+    const newEmail = await mailosaur.messages.get(
+      process.env.MAILOSAUR_SERVER_ID || "",
+      {
+        sentTo: testEmail,
+      }
+    );
+    expect(newEmail.subject).toBe("Verify your email");
+    const link = findLink(newEmail.html?.links || [], "Confirm my email");
+    expect(link?.text).toBe("Confirm my email");
+    const href = link?.href || "";
+    const page3 = await context3.newPage();
+    await page3.goto(href);
+    await page3.waitForSelector(
+      "text=Close this and return to your previous window",
+      {
+        timeout: 10000,
+      }
+    );
+    await page2.waitForTimeout(4000);
+    await page.goto(`${openloginURL}/wallet/apps`);
+    await page.reload();
+    await page.waitForURL(`${openloginURL}/wallet/apps`, {
+      waitUntil: "load",
+    });
+    await page.goto(`${openloginURL}/wallet/apps`);
+    expect(page.url()).toBe(`${openloginURL}/wallet/apps`);
+    await page.waitForURL(`${openloginURL}/wallet/apps`, {
+      waitUntil: "load",
+    });
+
+    expect(page.url()).toBe(`${openloginURL}/wallet/apps`);
+    expect(await page.isVisible("text=Authorized Apps")).toBeTruthy();
+    await page.waitForTimeout(4000);
+    expect(
+      await page.isVisible(
```

Comment:
We should have a basic assertion to verify that solana wallet app is visible in the list.

Reply:
added

Reply:
done